### PR TITLE
Implement `IN` and `NOT IN` string comparison function in Aim query

### PR DESCRIPTION
**Pull request by dsuhinin**

### CODE REVIEW

File: pkg/api/aim/query/query.go [hunk 1]
```diff
@@ -364,13 +342,36 @@ func (pq *parsedQuery) parseCompare(node *ast.Compare) (any, error) {
...
-				o, l, r, err := reverseComparison(op, left, right)
-				if err != nil {
-					return nil, err
-				}
-				exprs[i], err = newSqlComparison(o, l, r)
-				if err != nil {
-					return nil, err
+				switch op {
+				case ast.In:
+					return clause.Like{
```
Comment (jgiannuzzi) [inline pkg/api/aim/query/query.go:347]: we should probably check that `left` is a string in those 2 cases

**Review by jgiannuzzi — COMMENTED** [review]
I like it!!!

just a couple of comments below

File: pkg/api/aim/query/query.go (hunk 1 above)
Comment (dsuhinin) [reply pkg/api/aim/query/query.go:360]: done.

File: pkg/api/aim/query/query.go [hunk 2]
```diff
@@ -364,13 +342,36 @@ func (pq *parsedQuery) parseCompare(node *ast.Compare) (any, error) {
...
+						Value: fmt.Sprintf("%%%s%%", left),
+						Column: clause.Column{
+							Table: right.Table,
+							Name:  right.Name,
+						},
+					}, nil
+				case ast.NotIn:
+					return clause.NotConditions{
+						Exprs: []clause.Expression{
```
[the remark this answers is not in the document]
Comment (dsuhinin) [reply pkg/api/aim/query/query.go]: done.

File: pkg/api/aim/query/query.go [hunk 3]
```diff
@@ -38,12 +39,20 @@ type parsedQuery struct {
 	conditions []clause.Expression
 }
 
-type attributeGetter func(attr string) (any, error)
-
 type callable func(args []ast.Expr) (any, error)
 
+var negativeClause = func(expression clause.Expression) clause.Expression {
```
Comment (jgiannuzzi) [inline pkg/api/aim/query/query.go:44]: why do you make it a func variable and not directly a func? 🤔

File: pkg/api/aim/query/query.go [hunk 4]
```diff
@@ -344,25 +353,23 @@ func (pq *parsedQuery) parseCompare(node *ast.Compare) (any, error) {
 			case clause.Column:
 				switch op {
 				case ast.In:
+					// for `IN` statement, left parameter has to be always `string`.
+					if _, ok := left.(string); !ok {
+						return nil, eris.New("left parameter has to be a string")
```
Comment (jgiannuzzi) [inline pkg/api/aim/query/query.go:358]: why do you use `eris.New` instead of the stdlib `errors.New` here?

Comment (jgiannuzzi) [reply pkg/api/aim/query/query.go:358]: can you change all the `errors.New` and `fmt.Errorf` of this file to use `eris` then?

Comment (dsuhinin) [reply pkg/api/aim/query/query.go]: ok, I reverted to use `errors` instead. lets change this when we will have good unit tests for such important module.

File: pkg/api/aim/query/query.go (hunk 3 above)
Comment (dsuhinin) [reply pkg/api/aim/query/query.go]: changed in this style.

File: pkg/api/aim/query/query.go [hunk 5]
```diff
@@ -38,12 +38,12 @@ type parsedQuery struct {
...
 }
 
-type attributeGetter func(attr string) (any, error)
-
 type callable func(args []ast.Expr) (any, error)
 
 type subscriptSlicer func(index ast.Slicer) (any, error)
 
+type attributeGetter func(attr string) (any, error)
+
```
Comment (jgiannuzzi) [inline pkg/api/aim/query/query.go:46]: any reason why you moved `attributeGetter` here?

**Review by jgiannuzzi — COMMENTED** [review]
just one last nitpick, but otherwise LGTM 👍

File: pkg/api/aim/query/query.go (hunk 5 above)
Comment (jgiannuzzi) [reply pkg/api/aim/query/query.go:46]: hmmm, should it not be before `subscriptSlicer` then?

Comment (dsuhinin) [reply pkg/api/aim/query/query.go:46]: no no, not alphabetical - it is pain, just from short to long.

Comment (dsuhinin) [reply pkg/api/aim/query/query.go:46]: but yep, lets sort it fully.

Comment (jgiannuzzi) [reply pkg/api/aim/query/query.go:44]: but they have the same length for their names, and then `subscriptSlicer` has a longer signature — I'm cool with going with non-alphabetical and go for length, but we need to be consistent!

**Review by jgiannuzzi — APPROVED** [review]

File: pkg/api/aim/query/query.go (hunk 5 above)
Comment (dsuhinin) [reply pkg/api/aim/query/query.go:44]: yep, the use this way. agreed. done.

**Pull request merged**